frontend/config-assistant/src/components/gui-editor/properties/OneOfOfSelectionProperty.vue

```diff
@@ -48,6 +51,61 @@ const valueProperty = computed({
   },
 });
 
+onMounted(() => {
+  if (valueProperty.value === undefined && props.propertyData !== undefined) {
+    inferOneOfUserSelection();
+  }
+});
+
+function inferOneOfUserSelection() {
+  const pathAsString = pathToString(props.absolutePath);
+  const validationService = useSessionStore().validationService;
+
+  if (!useSessionStore().currentSelectedOneOfOptions.has(pathAsString)) {
+    // User has not yet made a oneOf sub-schema selection for this path
+    // --> auto infer a sub-schema
+
+    let index = 0;
+    for (let subSchema: JsonSchema of props.propertySchema.oneOf) {
+      const valid = validationService.validateSubSchema(
+        subSchema.jsonSchema!!,
+        // TODO: remove key once it has become optional (change from Paul).
+        pathToString(props.absolutePath) + '.oneOf[' + index + ']',
+        props.propertyData
+      ).valid;
+
+      if (valid) {
+        // found a oneOf subSchema for which the data is valid. Select it.
+        const optionToSelect = findOptionBySubSchemaIndex(index);
+        useSessionStore().currentSelectedOneOfOptions.set(pathAsString, optionToSelect!!);
+        return;
+      }
+
+      index++;
+    }
+  }
+}
+
+function findOptionByDisplayText(displayText: string): OneOfAnyOfSelectionOption | undefined {
+  for (let option of possibleOptions) {
```

Comment:
A map would have probably been more suited here but this is also fine (we don't have oneOfs with 1000s of options i think ;))